linera-chain/src/manager/multi.rs

```diff
@@ -307,12 +308,17 @@ impl MultiOwnerManager {
         key_pair: Option<&KeyPair>,
         now: Timestamp,
     ) {
+        let round = certificate.round;
+        // Validators only change their locked block if the new one is included in a proposal in the
+        // current round, or it is itself in the current round.
+        if key_pair.is_some() && round < self.current_round() {
+            return;
```

Comment:
Is this supposed to happen? Should we log an error?

Reply:
This could totally happen, even without any faults:

E.g. maybe the leader of round 1 collected enough `ValidatedBlock` signatures _just_ before the round timed out. So they send the certificate to the validators, but some validators are already in round 2.

Those will reject the certificate now, just in case the leader of round 2 was about to re-propose the block from round 0.